src/Schema/PostgreSQLSchemaManager.php

```diff
@@ -610,13 +610,50 @@ protected function selectTableNames(string $databaseName): Result
 
     protected function selectTableColumns(string $databaseName, ?string $tableName = null): Result
     {
+        
+        // feature detection: if attrgenerated exists it means we are 
+        // in a postgreSQL version supporting GENERATED columns
+        // see https://pgpedia.info/g/generated-column.html#implementation
+        $results = $this->_conn->executeQuery(<<<'SQL'
+            SELECT true
+            FROM information_schema.columns
+            WHERE table_name = 'pg_attribute'
+                AND column_name = 'attgenerated';
+        SQL);
+        $supportsGeneratedColumn = $results->fetchOne();
+
+        $fragmentDefaultValueSQL = <<<SQL
```

Comment:
I've decided to put the whole subquery rather than just the part that change (i.e the selected column )  because I felt it was easier for future maintainer to have a copy/pastable query rather than having to re-craft the query by hand when trying to understand what the code is doing

Reply:
```suggestion
        $fragmentDefaultValueSQL = <<<'SQL'
```